pyproject.toml

```diff
@@ -0,0 +1,120 @@
+[build-system]
+requires = ['hatchling', 'hatch-vcs']
+build-backend = 'hatchling.build'
+
+[project]
+name = 'fmf'
+authors = [
+    { name = 'Petr Splichal', email = 'psplicha@redhat.com' },
+]
+maintainers = [
+    { name = 'Petr Splichal', email = 'psplicha@redhat.com' },
+]
+description = 'Flexible Metadata Format'
+readme = 'README.rst'
+license = 'GPL-2.0-or-later'
+license-files = { paths = ['LICENSE'] }
+requires-python = '>=3.9'
+classifiers = [
+    'Natural Language :: English',
+    'Programming Language :: Python :: 3.9',
+    'Programming Language :: Python :: 3.10',
+    'Programming Language :: Python :: 3.11',
+    'Programming Language :: Python :: 3.12',
+    'Topic :: Utilities',
+]
+keywords = [
+    'metadata',
+    'testing',
+]
+dependencies = [
+    'ruamel.yaml',
+    'filelock',
+    'jsonschema',
+]
+dynamic = ['version']
+
+[project.urls]
+Homepage = 'https://github.com/teemtee/fmf'
+
+[project.optional-dependencies]
+tests = [
+    'pytest',
+]
+tests-cov = [
+    'fmf[tests]',
+    'pytest-cov',
+]
+dev = [
+    'fmf[tests]',
+    'pre-commit',
+]
+docs = [
+    'sphinx',
+    'sphinx_rtd_theme',
+]
+all = [
+    'fmf[dev]',
+    'fmf[docs]',
+]
+
+[project.scripts]
+fmf = 'fmf.cli:cli_entry'
+
+[tool.hatch]
+version.source = 'vcs'
+version.raw-options.version_scheme = 'post-release'
```

Comment:
I think that one was from `packit`'s configuration. I do prefer `guess-dev` personally if everyone is ok with it.

Reply:
I like the default. Note that in tmt, @psss had to do some magic with hardcoding dev versions in development copr rpms. See https://github.com/teemtee/tmt/blob/ff08000eaccfd5a02a54c5e8f4e1394f6aa662a7/.packit.yaml#L32

Reply:
Oh interesting. I usually have spec files always use the git version so that it doesn't use the actual dev package version like:
```yaml
update_release: false
upstream_tag_template: v{version}

jobs:
  - &copr_build
    job: copr_build
    trigger: release
    owner: "@scikit-build"
    project: release
  - <<: *copr_build
    trigger: commit
    branch: main
    project: nightly
  - <<: *copr_build
    trigger: pull_request
    project: scikit-build-core
    update_release: true
    release_suffix: "{PACKIT_RPMSPEC_RELEASE}"
```
It worked well for `scikit-build-core`, e.g. [copr repo](https://copr.fedorainfracloud.org/groups/g/scikit-build/coprs/). Only for the PR does it have the weird format like: `0.9.2-1.20240424024056580646.pr725.3.g400b085`, but it shouldn't matter there.